src/wallet/wallet.js

```diff
@@ -53,12 +59,22 @@ export class Wallet {
       throw new Error('Hex encoded session key expected.')
     }
 
+    keypairs = Array.from([
```

Comment:
all you do at lines 30-40 looks related to what you do here. why then place both actions in such different places of method?